### DISCUSSION TITLE
Using libraries installed by brew with clang

### DISCUSSION THREAD

**Discussion by nwbreneman** [Homebrew/discussions#3792] [Everyday usage]
Hello,

As a new brew user on macOS (and a new macOS user), I installed `libssh` and `libssh2` to fulfill requirements for the Python package `ansible-pylibssh`. The package installation failed during a clang build due to not being able to find `libssh` installed via brew. I was able to resolve by exporting the CFLAGS variable with the correct paths: `CFLAGS=-I /opt/homebrew/include -I ext -L /opt/homebrew/lib -lssh`. I'm just wondering if this is expected behavior for using such dependencies from brew or if it's considered a bug. Thanks!

Here's the clang error for reference:
```
        clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include -I/Users/NBreneman/code/atlantis-integration-test/.venv/include -I/Users/NBreneman/.asdf/installs/python/3.8.12/Library/Frameworks/Python.framework/Versions/3.8/include/python3.8 -c /private/var/folders/g4/mmmhnwrn5ng8s2d6lmhm72h40000gq/T/pip-req-build-aflyuh3p/src/pylibsshext/_libssh_version.c -o /private/var/folders/g4/mmmhnwrn5ng8s2d6lmhm72h40000gq/T/pip-req-build-aflyuh3p/src/tmpqg5vzenr/private/var/folders/g4/mmmhnwrn5ng8s2d6lmhm72h40000gq/T/pip-req-build-aflyuh3p/src/pylibsshext/_libssh_version.o
        /private/var/folders/g4/mmmhnwrn5ng8s2d6lmhm72h40000gq/T/pip-req-build-aflyuh3p/src/pylibsshext/_libssh_version.c:757:10: fatal error: 'libssh/libssh.h' file not found
        #include "libssh/libssh.h"
                 ^~~~~~~~~~~~~~~~~
        1 error generated.
        error: command '/usr/bin/clang' failed with exit code 1
```

**Comment by boredcoder411** [comment]
Did you ever get this to work?

**Reply by nwbreneman** [reply]
I don't remember if I did anything else after this to be honest. Exporting the CFLAGS environment variable before doing the install did fix it for me then. I haven't tried a fresh install since this.
